sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SwaggerMethodParser.java

```diff
@@ -428,6 +437,31 @@ public Object setBody(Object[] swaggerMethodArguments) {
         return result;
     }
 
+    /**
+     * Generate the object to be used as the value of the HTTP request.
+     *
+     * @param swaggerMethodArguments The method arguments to generate the value object from.
+     * @return The object that will be used as the body of the HTTP request
+     */
+    public String setMultipartBody(Object[] swaggerMethodArguments, String boundary) {
+        String result = null;
+
+        if (!CoreUtils.isNullOrEmpty(formDataEntries) && swaggerMethodArguments != null) {
+            String boundaryPrefix = "--" + boundary + System.lineSeparator();
+            String boundaryDelimiter = System.lineSeparator() + "--" + boundary + System.lineSeparator();
+            String boundarySuffix = System.lineSeparator() + "--" + boundary + "--";
+
+            result = formDataEntries.stream()
+                .map(formDataEntry -> serializeMultipartFormData(serializer, formDataEntry.getFormData().value(),
+                    swaggerMethodArguments[formDataEntry.getParameterIndex()],
+                    formDataEntry.getFormData().filename()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.joining(boundaryDelimiter, boundaryPrefix, boundarySuffix));
```

Comment:
I suspect a more traditional `for` loop with a `StringBuilder` may be more performant here? It might also help to minimise the amount of GC churn with `StringBuilder` instances being created in the `serializeMultipartFormData` method.

Reply:
If you think this would be more performant then we should consider doing the same for other methods like `setBody()` when we deal with `FormData`, as that also uses a similar approach of `stream()` + `Collectors.joining()`, while the internal `serializeFormData()` does the same (`stream()` + `Collectors.joining()`).

Reply:
I did make the change though :)

Reply:
Thanks - looks much better to me. We should definitely do similar wherever appropriate to reduce GC churn.